src/Generator/Driver.cs

```diff
@@ -353,66 +350,55 @@ private void WriteGeneratedCodeToFile(string file, string generatedCode)
 
         private static readonly Dictionary<Module, string> libraryMappings = new Dictionary<Module, string>();
 
-        public void CompileCode(Module module)
+        public bool CompileCode(Module module)
         {
-            var assemblyFile = Path.Combine(Options.OutputDir, module.LibraryName + ".dll");
-
-            var docFile = Path.ChangeExtension(assemblyFile, ".xml");
-
-            var compilerOptions = new StringBuilder();
-            compilerOptions.Append($" /doc:\"{docFile}\"");
-            compilerOptions.Append(" /debug:pdbonly");
-            compilerOptions.Append(" /unsafe");
-
-            var compilerParameters = new CompilerParameters
-            {
-                GenerateExecutable = false,
-                TreatWarningsAsErrors = false,
-                OutputAssembly = assemblyFile,
-                GenerateInMemory = false,
-                CompilerOptions = compilerOptions.ToString()
-            };
-
-            if (module != Options.SystemModule)
-                compilerParameters.ReferencedAssemblies.Add(
-                    Path.Combine(Options.OutputDir, $"{Options.SystemModule.LibraryName}.dll"));
-            // add a reference to System.Core
-            compilerParameters.ReferencedAssemblies.Add(typeof(Enumerable).Assembly.Location);
-
             var location = System.Reflection.Assembly.GetExecutingAssembly().Location;
-            var outputDir = Path.GetDirectoryName(location);
-            var locationRuntime = Path.Combine(outputDir, "CppSharp.Runtime.dll");
-            compilerParameters.ReferencedAssemblies.Add(locationRuntime);
-
-            compilerParameters.ReferencedAssemblies.AddRange(
-                (from dependency in module.Dependencies
-                 where libraryMappings.ContainsKey(dependency)
-                 select libraryMappings[dependency]).ToArray());
-
-            compilerParameters.ReferencedAssemblies.AddRange(module.ReferencedAssemblies.ToArray());
-
-            Diagnostics.Message($"Compiling {module.LibraryName}...");
-            CompilerResults compilerResults;
-            using (var codeProvider = new CSharpCodeProvider(
-                new Dictionary<string, string> {
-                    { "CompilerDirectoryPath", ManagedToolchain.FindCSharpCompilerDir() } }))
+            string csproj = Path.Combine(Options.OutputDir, $"{module.LibraryName}.csproj");
+            File.WriteAllText(Path.Combine(Options.OutputDir, "Directory.Build.props"), "<Project />");
+            File.WriteAllText(csproj,
```

Comment:
Please move this out of the driver into its own generator.

Reply:
Could you explain, please? Do you want me to move the code to an external function? I don't get what generator you want. The only generators I'm aware of are all for source code and they have nothing to do with this piece because it doesn't work with the AST.